x86_64.c

```diff
@@ -6460,8 +6465,10 @@ x86_64_ORC_init(void)
 
 	orc->has_signal = MEMBER_EXISTS("orc_entry", "signal");	/* added at 6.3 */
 	orc->has_end = MEMBER_EXISTS("orc_entry", "end");	/* removed at 6.4 */
-
-	if (orc->has_signal && !orc->has_end)
+	if (try_get_symbol_data("orc_header", sizeof(orc_header), orc_header) &&
+	    memcmp(orc_header, orc_hash_6_4, 20) == 0) {
```

Comment:
We might want to move these lines since we have checks on lines 6413-6421 that can fail.